src/azul/azulclient.py

```diff
@@ -228,29 +227,32 @@ def notifications_queue(self):
 
     def remote_reindex(self,
                        catalog: CatalogName,
-                       partition_prefix_length: int,
                        sources: AbstractSet[str]):
-        partition_prefixes = [
-            ''.join(partition_prefix)
-            for partition_prefix in product('0123456789abcdef',
-                                            repeat=partition_prefix_length)
-        ]
-
-        def message(source: str, partition_prefix: str) -> JSON:
-            logger.info('Remotely reindexing prefix %r of source %r into catalog %r',
-                        partition_prefix, source, catalog)
-            return dict(action='reindex',
-                        catalog=catalog,
-                        source=str(source),
-                        prefix=partition_prefix)
 
-        messages = starmap(message, product(sources, partition_prefixes))
-        for batch in chunked(messages, 10):
-            entries = [
-                dict(Id=str(i), MessageBody=json.dumps(message))
-                for i, message in enumerate(batch)
-            ]
-            self.notifications_queue.send_messages(Entries=entries)
+        plugin = self.repository_plugin(catalog)
+        for source in sources:
+            source_name = plugin.resolve_source(name=source).name
+
+            def message(partition_prefix: str) -> JSON:
+                logger.info('Remotely reindexing prefix %r of source %r into catalog %r',
+                            partition_prefix, source, catalog)
+                return dict(action='reindex',
+                            catalog=catalog,
+                            source=str(source),
+                            prefix=partition_prefix)
+
+            partition_prefix = [''.join(partition_prefix)
+                                for partition_prefix in product('0123456789abcdef',
+                                                                repeat=source_name.partition_prefix_length)
+                                ]
```

Comment:
Nice job reorganizing this section, but the wrapping here isn't right:
```suggestion
            partition_prefix = [
                ''.join(partition_prefix)
                for partition_prefix in product('0123456789abcdef',
                                                               repeat=source_name.partition_prefix_length)
            ]
```

That said, this could easily be a generator:

```suggestion
            partition_prefix = map(''.join, product('0123456789abcdef',
                                                                       repeat=source_name.partition_prefix_length))
```

And I also can't help but feel that the variable name should be plural:

```suggestion
            partition_prefixes = ...
```

But I might be being a little overzealous here. Your call.

Reply:
actually the mapping on the `partition_prefixes` is a good call, if the length is 4, then the length of the list is 65536 values.... so def an improvement to use the generator.

It should have been plural in the first place.